### PR TITLE
feat(ui): Move API Applications to new settings

### PR DESCRIPTION
**Pull request by billyvg**
Depends on ~#6885~, ~#6880~, ~#6879~, ~#6872~

Polish would be nice:
image

Need to style the "Credentials" inputs to make them look disabled. Also do we strictly rely on breadcrumbs + nav bar for navigation (i.e. do we want a Cancel button or something)?
image

### CODE REVIEW

File: src/sentry/static/sentry/app/views/settings/account/apiApplicationDetails.jsx [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+    });
+  }
+
+  onRemoveApplication(app) {}
+
+  getTitle() {
+    return 'Application Details - Sentry';
+  }
+
+  handleSubmitSuccess = (change, model, id) => {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/account/apiApplicationDetails.jsx]: This is duplicated code, needs to be refactored.

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by billyvg** [comment]
Note tests are currently broken because it needs to be rebased with deps.

**Comment by MaxBittker** [comment]
the create application button should probably be lifted off of the line so it doesn't touch

File: src/sentry/static/sentry/app/data/forms/apiApplication.jsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
+// import {createSearchMap} from './util';
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/data/forms/apiApplication.jsx]: why doesn't this go into the search map? should the comment be removed?

Comment (billyvg) [reply src/sentry/static/sentry/app/data/forms/apiApplication.jsx]: Seems weird to have the form in search, I think when you're searching you'll be searching for the route and not a specific settings to change?

Comment (billyvg) [reply src/sentry/static/sentry/app/data/forms/apiApplication.jsx]: But maybe I'm wrong

**Comment by billyvg** [comment]
@Chrissy can you help style the TextCopyInput component (its on master)? icon looks funny + input should have a "readonly" look to it

**Comment by Chrissy** [comment]
@billyvg cha!

**Comment by Chrissy** [comment]
styled component is here https://github.com/getsentry/sentry/pull/6997

File: src/sentry/static/sentry/app/routes.jsx [hunk 3]
```diff
@@ -164,6 +164,23 @@ const accountSettingsRoutes = [
...
+        componentPromise={() =>
+          import(/*webpackChunkName: "ApiApplications"*/ './views/settings/account/apiApplications')}
+        component={errorHandler(LazyLoad)}
+      />
+      <Route
+        path=":appId/"
+        name="Details"
+        componentPromise={() =>
+          import(/*webpackChunkName: "ApiApplicationDetails"*/ './views/settings/account/apiApplicationDetails')}
+        component={errorHandler(LazyLoad)}
```
Comment (MaxBittker) [inline src/sentry/static/sentry/app/routes.jsx]: is this being chosen to be lazy loaded based on size or mostly infrequency?

**Review by MaxBittker — APPROVED** [review]
looks good to me

File: src/sentry/static/sentry/app/routes.jsx (hunk 3 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/routes.jsx]: Mostly infrequency, (but atm could be b/c of size if it uses forms)

**Pull request merged**